opal/corelib/runtime.js

```diff
@@ -1350,7 +1350,7 @@
       }
     }
 
-    if (obj.$method_added && !obj.$method_added.$$stub) {
+    if (obj.$method_added && !obj.$method_added.$$stub && (obj.$$singleton_of !== Opal.BasicObject)) {
```

Comment:
In light of what pointed out by @eregon in https://github.com/ruby/spec/pull/200#issuecomment-190636009 this probably should look like 
```rb
if (obj.$method_added && !obj.$method_added.$$stub && !obj.$$singleton_of) {
  …
}
else if (obj.$singleton_method_added && !obj.$singleton_method_added.$$stub && obj.$$singleton_of) {
  …
}
```

and that would probably fix some more ruby specs too… :)